src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -235,9 +232,13 @@ private void importRoslynReport(String reportPath) {
       throw Throwables.propagate(e);
     }
 
-    Set<String> activeRuleIds = Sets.newHashSet();
-    for (ActiveRule activeRule : ruleProfile.getActiveRulesByRepository(CSharpPlugin.REPOSITORY_KEY)) {
-      activeRuleIds.add(activeRule.getRuleKey());
+    ImmutableMultimap<String, ActiveRule> activeRoslynRulesByPluginKey = RoslynProfileExporter.activeRoslynRulesByPluginKey(ruleProfile.getActiveRules());
+    Map<String, String> repositoryKeyByRoslynRuleKey = Maps.newHashMap();
+    for (ActiveRule activeRoslynRule: activeRoslynRulesByPluginKey.values()) {
+      String previousRepositoryKey = repositoryKeyByRoslynRuleKey.put(activeRoslynRule.getRuleKey(), activeRoslynRule.getRepositoryKey());
+      if (previousRepositoryKey != null) {
+        throw new IllegalArgumentException("Rule keys must be unique, but \"" + activeRoslynRule.getRuleKey() + "\" is defined in both the \"" + previousRepositoryKey + "\" and \"" + activeRoslynRule.getRepositoryKey() + "\" rule repositories.");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Split this 246 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00103)